packages/devtools_app/lib/src/screens/performance/panes/queued_microtasks/queued_microtasks_view.dart

```diff
@@ -0,0 +1,278 @@
+          dataKey: 'queued-microtask-selector',
+          columns: [_idColumn],
+          defaultSortColumn: _idColumn,
+          defaultSortDirection: SortDirection.ascending,
+          onItemSelected: (indexedMicrotask) =>
+              _setSelectedMicrotask(indexedMicrotask?.$2),
+        ),
+      ),
+    ],
+  );
+}
+
+class StackTraceView extends StatelessWidget {
+  const StackTraceView({super.key, required selectedMicrotask})
+    : _selectedMicrotask = selectedMicrotask;
+
+  final Microtask? _selectedMicrotask;
+
+  @override
+  Widget build(BuildContext context) {
+    final theme = Theme.of(context);
+
+    return Column(
+      children: [
+        SizedBox.fromSize(
+          size: Size.fromHeight(defaultHeaderHeight),
+          child: Container(
+            decoration: BoxDecoration(
+              border: Border(bottom: defaultBorderSide(theme)),
+            ),
+            padding: const EdgeInsets.only(left: defaultSpacing),
+            alignment: Alignment.centerLeft,
+            child: const Row(
+              children: [
+                Text('Stack trace captured when microtask was enqueued'),
+              ],
+            ),
+          ),
+        ),
+        Row(
+          children: [
+            Expanded(
+              child: Padding(
+                padding: const EdgeInsets.symmetric(
+                  vertical: denseRowSpacing,
+                  horizontal: defaultSpacing,
+                ),
+                child: SelectableText(
+                  style: theme.fixedFontStyle,
+                  _selectedMicrotask!.stackTrace.toString(),
+                ),
+              ),
+            ),
+          ],
+        ),
+      ],
+    );
+  }
+}
+
+class QueuedMicrotasksTabView extends StatefulWidget {
+  const QueuedMicrotasksTabView({super.key, required this.controller});
+
+  final QueuedMicrotasksController controller;
+
+  @override
+  State<QueuedMicrotasksTabView> createState() =>
+      _QueuedMicrotasksTabViewState();
+}
+
+class _QueuedMicrotasksTabViewState extends State<QueuedMicrotasksTabView>
+    with AutoDisposeMixin {
+  static final _dateTimeFormat = DateFormat('HH:mm:ss.SSS (MM/dd/yy)');
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder(
+      valueListenable: widget.controller.status,
+      builder: (context, status, _) {
+        if (status == QueuedMicrotasksControllerStatus.empty) {
+          return const RefreshQueuedMicrotasksInstructions();
+        } else if (status == QueuedMicrotasksControllerStatus.refreshing) {
+          return Center(
+            child: Text(
+              style: Theme.of(context).regularTextStyle,
+              'Refreshing...',
+            ),
+          );
+        } else {
+          return ValueListenableBuilder(
+            valueListenable: widget.controller.queuedMicrotasks,
+            builder: (_, queuedMicrotasks, _) {
+              assert(queuedMicrotasks != null);
```

Comment:
after this assert, should we be defensive and return `CenteredMessage('Unexpected null value')` or similar?